pyproject.toml

```diff
@@ -11,6 +11,8 @@ dependencies = [
     "ai-exchange>=0.8.0",
     "click>=8.1.7",
     "prompt-toolkit>=3.0.47",
+    "sentence-transformers>=3.0.1",
```

Comment:
Maybe we include these as optional dependencies instead of required. The reason being that they are pretty  heavyweight and only needed for this toolkit. So something like this: https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#a-full-example

Reply:
@zakiali what happens when they are optional - and you enable this in toolkits - will it bring them in then? as that wouldn't be ideal for the user (as goose is a cli not a lib?), I guess maybe this is weakness of using py plugins for toolkits (I think there may be better ways). In this case, it *may* be justified (the real heavyness happens when it loads the toolkit) as this seems needed for large codebases

Reply:
have made optional